src/main/java/net/masterthought/cucumber/json/Match.java

```diff
@@ -1,13 +1,82 @@
 package net.masterthought.cucumber.json;
 
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.LinkedList;
+import java.util.List;
+
 public class Match {
 
     // Start: attributes from JSON file report
-    private final String location = null;
+    private final JsonElement location = null;
     // End: attributes from JSON file report
 
     public String getLocation() {
-        return location;
+        if (location == null) {
+            return null;
+        }
+        StringBuilder sb = new StringBuilder();
+        JsonObject locationObject;
+
+        if (location.isJsonPrimitive() && location.getAsJsonPrimitive().isString()) {
+            return location.getAsString();
+        } else {
+            locationObject = location.getAsJsonObject();
+            String filename;
+            if (locationObject.has("file") && locationObject.get("file").getAsJsonPrimitive().isString()) {
+                filename = locationObject.get("file").getAsString();
+            } else if (locationObject.has("filepath") && locationObject.get("filepath").isJsonObject()) {
+                JsonObject filePath = locationObject.get("filepath").getAsJsonObject();
+                if (filePath.has("filename") && filePath.get("filename").getAsJsonPrimitive().isString()) {
+                    filename = filePath.get("filename").getAsString();
+                } else {
+                    return null;
+                }
+            } else {
+                return null;
+            }
+
+            sb.append(filename).append(":");
+
+            LinkedList<Integer> lineList = new LinkedList<>();
+            if (locationObject.has("lines") && locationObject.get("lines").isJsonObject()) {
+                JsonObject lines = locationObject.get("lines").getAsJsonObject();
+                if (lines.has("data") && lines.get("data").isJsonArray()) {
+                    JsonArray data = lines.get("data").getAsJsonArray();
+                    for (JsonElement lineElement : data) {
+                        try {
+                            int line = Integer.parseInt(lineElement.getAsString());
+                            lineList.add(line);
+                        } catch (NumberFormatException e) {
+                            return null;
+                        }
+                    }
+                } else {
+                    return null;
+                }
+            } else {
+                return null;
+            }
+            ;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [An empty statement (semicolon) not part of a loop](https://www.codacy.com/app/damianszczepanik/cucumber-reporting/file/2048973241/issues/source?bid=3059308&fileBranchId=3097149#l63)